plugins/pi4j-plugin-gpiod/src/main/java/com/pi4j/plugin/gpiod/provider/gpio/digital/GpioDDigitalInputProviderImpl.java

```diff
@@ -46,6 +46,11 @@ public int getPriority() {
     @Override
     public DigitalInputProvider initialize(Context context) throws InitializeException {
         DigitalInputProvider provider = super.initialize(context);
+        // Retrieve the chip name from the context properties
+        String chipName = context.properties().get("gpio.chip.name");
+        if (chipName != null) {
+            GpioDContext.getInstance().setChip(chipName);
+        }
```

Comment:
I think it would make more sense to pass the properties to the initialize() method, in this way, we only need the property name once.